packages/superset-ui-legacy-preset-chart-big-number/src/BigNumber/transformProps.ts

```diff
@@ -0,0 +1,132 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import * as color from 'd3-color';
+import { getNumberFormatter, NumberFormats } from '@superset-ui/number-format';
+import { ChartProps } from '@superset-ui/chart';
+import getTimeFormatterForGranularity from '../utils/getTimeFormatterForGranularity';
+
+const TIME_COLUMN = '__timestamp';
+
+export default function transformProps(chartProps: ChartProps) {
+  const { width, height, formData, queryData } = chartProps;
+  const {
+    colorPicker,
+    compareLag: compareLagInput,
+    compareSuffix = '',
+    headerFontSize,
+    metric,
+    showTrendLine,
+    startYAxisAtZero,
+    subheader = '',
+    subheaderFontSize,
+    timeGrainSqla: granularity,
+    vizType,
+    timeRangeFixed = false,
+    timeRangeUseFallback = false,
```

Comment:
Perhaps rename this field and change from `boolean` to `bigNumberMode: 'last' | 'lastAvailable'`.
We could extend this to support additional modes: `'average'`, `'sum'` in the future.

Reply:
In that case, should we use a `SelectControl` in Superset, too? I'm a little hesitate to add this for now. Allowing `sum`, `average` or any other computation mode significantly increases the complexity of how the product can be used and may cause confusion among end users. Even if the chart builders know what they are doing, the readers might not.

Reply:
I am just afraid to add this `timeRangeUseFallback` field and soon have to write `db:migration` for it. 
At least, could rename it to `useLastAvailableDataPoint` to be more specific.

For more context: 

The current `BigNumber` and `BigNumberTotal` are behaving differently. One show last value and one show sum. 
There was a feature request to clarify this at one point. 

The returned data from the legacy API are also different.

`BigNumber`: array of value over time

```
[
  {
    __timestamp: 1388534400000.0,
    sum__SP_POP_TOTL: 7237260256.0,
  },
  {
    __timestamp: 1356998400000.0,
    sum__SP_POP_TOTL: 7151135481.0,
  },
  {
    __timestamp: 1325376000000.0,
    sum__SP_POP_TOTL: 7066007165.0,
  },
  {
    __timestamp: 1293840000000.0,
    sum__SP_POP_TOTL: 6984252419.0,
  }
]
```

`BigNumberTotal`: sum as single entry in the array

```
[
  {
    sum__num: 32546308,
  },
]
```

Reply:
Hmm. Why does this sound intuitive and surprising at the same time... 😂 Still, I worry it will be probably even more confusing if Big Number with Trend Line can display sums, too.

Reply:
Another alternative solution is to fallback to last available data point by default, but add warning to indicate that the last data point is missing.

Reply:
options:
1. current solution but rename variable from `timeRangeUseFallback` to `useLastAvailableDataPoint` (or something that is more descriptive).
2. fallback to last available data point by default, but add warning to indicate that the last data point is missing. (maybe your current solution 1. is already better, or can circle back with @etr2460)
3. changes from boolean to string. 

I'm good with any one them.

Reply:
I like your alternative solution. Maybe display the Big Number grayed out and add a tooltip explaining this is the last available value. It saves the trouble for future migration and makes sure chart readers understand what the value is, too.

Reply:
![image](https://user-images.githubusercontent.com/335541/77501534-8f8d6b80-6e15-11ea-9e4b-15aa6a44fee8.png)

Come up with this: fallback to the last available value by default + add some hover tooltip to the fallback values.